db/comdb2.c

```diff
@@ -807,6 +822,18 @@ struct dbtable *get_dbtable_by_name(const char *p_name)
     return p_db;
 }
 
+unsigned long long get_next_genid_for_table(const char *tablename)
+{
+    struct dbtable *tbl = NULL;
+    unsigned long long genid = 0;
+    Pthread_rwlock_rdlock(&thedb_lock);
+    tbl = hash_find_readonly(thedb->db_hash, &tablename);
+    if (tbl)
+        genid = bdb_get_next_genid(tbl->handle);
```

Comment:
Is there a case where tbl is valid but tbl->handle is not? Maybe thedb_lock is not enough to protect the handle from being swapped (i.e. finalize schema change).. However, I do notice that the underlying function calls only use bdb_state->parent/bdb_state->attr which should not change even during swap. So it might be fine but it worth double checking.